File: Models/Trend.cs

namespace Project1.Models
{
    public class Trend
    {
        //public int? Id { get; set; }
        public Variable? Variable { get; set; }
        public TrendPoint[]? Data { get; set; }
        public DateTime? Start { get; set; }
        public DateTime? End { get; set; }
    }
}

File: ClientApp/src/app/trends/chart-trends/trend.ts
import { TrendPoint } from './trend-point';

export class Trend {
  name: string;
  color: string;
  data: TrendPoint[];

  constructor(name: string, color: string, data: TrendPoint[]) {
    this.name = name;
    this.color = color;
    this.data = data;
  }
}
